src/pip/_internal/wheel.py

```diff
@@ -1036,14 +1037,22 @@ def build(
         # type: (...) -> List[InstallRequirement]
         """Build wheels.
 
-        :param unpack: If True, replace the sdist we built from with the
-            newly built wheel, in preparation for installation.
+        :param autobuilding: If True, replace the sdist we built from with the
+            unpacked newly built wheel, in preparation for installation.
         :return: True if all the wheels built correctly.
         """
```

Comment:
Now that you've changed the test to fix all call sites, how about adding `assert self.wheel_cache` to the beginning?

Reply:
I did not do it so far because I figured it was obvious enough that wheel_cache was a mandatory argument of the WheelBuilder constructor (per the typing declaration), and in case one makes the error, the crash reason will be as easy to understand as an assert. So I thought an additional assert about that would have little additional information value.

Reply:
> So I thought an additional assert about that would have little additional information value.

I wouldn't say it adds little value as I do think it helps people reading the code. (Case in point: it would have answered [@pradyunsg's question](https://github.com/pypa/pip/pull/6869#discussion_r313374668).) But I agree it's redundant with the type annotation, so I'm okay with leaving it out.

Reply:
> So I thought an additional assert about that would have little additional information value.

I wouldn't say it adds little value as I do think it helps people reading the code. (Case in point: it would have answered [@pradyunsg's question](https://github.com/pypa/pip/pull/6869#discussion_r313374668).) But I agree it's redundant with the type annotation, so I'm okay with leaving it out.